src/components/main/compose/msg/mod.rs

```diff
@@ -235,6 +236,58 @@ pub fn Msg<'a>(cx: Scope<'a, Props>) -> Element<'a> {
                                 onpressed: move |_| popout.set(true),
                                 text: String::from("Reply"),
                                 icon: Shape::ArrowUturnLeft,
+                            },
+                            ContextItem {
+                                onpressed: move |_| {
+                                    let state = use_atom_ref(&cx, STATE).clone();
+                                    let rg = cx.props.messaging.clone();
+                                    let conversation_id = cx.props.message.clone().conversation_id();
+                                    cx.spawn({
+                                        to_owned![rg, conversation_id, state];
+                                        async move {
+                                            match rg.delete(conversation_id, None).await {
+                                                Ok(_) => {
+                                                    state.write().dispatch(Actions::RemoveConversation(conversation_id));
+                                                    log::info!("successfully delete conversation")
+                                                },
+                                                Err(error) => log::error!("error when deleting conversation: {error}"),
+                                            };
+                                        }
+                                    });
+                                },
+                                text: String::from("Clear Chat"),
+                                danger: true,
+                                icon: Shape::Trash,
+                            },
+                            ContextItem {
+                                onpressed: move |_| {
+                                    let state = use_atom_ref(&cx, STATE).clone();
+                                    let rg = cx.props.messaging.clone();
+                                    let conversation_id = cx.props.message.clone().conversation_id();
+                                    cx.spawn({
+                                        to_owned![rg, conversation_id, state];
+                                        async move {
+                                            match rg.delete(conversation_id, None).await {
+                                                Ok(_) => {
+                                                    state.write().dispatch(Actions::RemoveConversation(conversation_id));
```

Comment:
i believe that per @dariusc93 's comment, this dispatch should happen after line 282.

Reply:
You are correct.